client/src/pages/Home.svelte

```diff
@@ -1,17 +1,26 @@
-<script>
+<script lang="ts">
     import Button from '../components/ui/Button.svelte';
-
+    import { goToTrackingPage } from '../components/ui/itemrow/util.ts';
     import { register } from './register.ts';
-    import { ButtonType, InputType, IconColor } from '../components/types.ts';
+    import { ButtonType, IconColor } from '../components/types.ts';
 
     import Google from '../components/icons/Google.svelte';
     import Camera from '../components/icons/Camera.svelte';
     import Search from '../components/icons/Search.svelte';
     import TextInput from '../components/ui/TextInput.svelte';
+    import Modal from '../components/ui/Modal.svelte';
+    import QrScanner from '../components/ui/QRScanner.svelte';
+    import { assert } from '../assert.ts';
 
     const placeholderSrc = new URL('../assets/images/logo-background.png', import.meta.url);
+    let showScan = false as boolean;
 
     let trackingNumber = '';
+
+    function scanHandler(e: CustomEvent) {
+        assert(typeof e.detail === 'string' );
+        goToTrackingPage(e.detail);
+    }

```

Comment:
I believe this should be allowed, right?
```suggestion
    function scanHandler({ detail }: CustomEvent<string>) {
        goToTrackingPage(detail);
    }
```